code/modules/mob/living/emote.dm

```diff
@@ -446,25 +448,47 @@
 		to_chat(user, "<span class='boldwarning'>You cannot send IC messages (muted).</span>")
 		return FALSE
 	else if(!params)
-		var/custom_emote = copytext(sanitize(input("Choose an emote to display.") as text|null), 1, MAX_MESSAGE_LEN)
+		custom_emote = copytext(sanitize(input("Choose an emote to display.") as text|null), 1, MAX_MESSAGE_LEN)
 		if(custom_emote && !check_invalid(user, custom_emote))
 			var/type = input("Is this a visible or hearable emote?") as null|anything in list("Visible", "Hearable")
 			switch(type)
 				if("Visible")
-					emote_type = EMOTE_VISIBLE
+					custom_emote_type = EMOTE_VISIBLE
 				if("Hearable")
-					emote_type = EMOTE_AUDIBLE
+					custom_emote_type = EMOTE_AUDIBLE
 				else
 					tgui_alert(usr,"Unable to use this emote, must be either hearable or visible.")
 					return
-			message = custom_emote
 	else
-		message = params
+		custom_emote = params
 		if(type_override)
-			emote_type = type_override
-	. = ..()
```

Comment:
You should not be replacing this parent call with what seems to be copy pasted code from the parent. Work the changes you need into the base if possible.

Reply:
I know that I probably shouldn't be doing that, but I didn't want to change how every single other emote worked, by introducing another variable that they wouldn't need, and that only *me would require, as every single other emote has an hard-coded message, so even if the message technically gets overwritten, it doesn't matter because it's overwritten by the exact same message.

I also wasn't sure if modifying the parent proc wouldn't cause some problems with the other emotes, which was something I wanted to avoid.

What would you suggest I do to the base proc to avoid issues, then? Do the same type of juggling with a local variable?

Reply:
@ninjanomnom

Reply:
We talked about this in discord and I'll probably be working with them on a more comprehensive refactor later